### PR TITLE
feat: setData API

### PR DESCRIPTION
**Pull request by jung-han**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

```js
chart.setData(data);
```

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/component/component.ts [hunk 1]
```diff
@@ -244,6 +222,50 @@ export default abstract class Component {
...
+    modelNames,
+    targetNames
+  ) {
+    if (currentModels.length > targetModels.length) {
+      const newModels = models.filter(({ name }) => includes(targetNames, name));
+
+      return newModels.length !== targetModels.length ? targetModels : newModels;
+    }
+
+    if (currentModels.length < targetModels.length) {
```
Comment (lja1018) [inline src/component/component.ts:238]: `currentModels`랑 `targetModels`의 length를 빼서 쓰면 어떨까요?

**Review by lja1018 — APPROVED** [review]
[12/03] 리뷰완료
고생하셨습니다!

File: src/store/treemapSeriesData.ts [hunk 2]
```diff
@@ -102,10 +102,7 @@ const treemapSeriesData: StoreModule = {
   }),
   action: {
     setTreemapSeriesData({ state }) {
-      extend(
-        state.treemapSeries,
-        makeTreemapSeries(state.series, state.options as TreemapChartOptions)
-      );
+      state.treemapSeries = makeTreemapSeries(state.series, state.options as TreemapChartOptions);
```
Comment (jungeun-cho) [inline src/store/treemapSeriesData.ts:105]: 해당 파일 상단에 import된 extend 삭제요~

**Review by jungeun-cho — APPROVED** [review]
[12/03] 이견없습니다. 수고하셨어요!

**Pull request merged**